### ISSUE TITLE
[5.7] Tests using the be() and/or actingAs() helpers fail

### ISSUE THREAD

**Issue by stefpankov** [laravel/lumen-framework#808]
- Lumen Version: 5.7.0
- PHP Version: 7.2

### Description:
Tests using the be() and/or actingAs() helpers but not making an actual request fail with the following message:

> InvalidArgumentException: Auth driver [api] for guard [api] is not defined.

After investigating the issue, I concluded that it's due to the newest changes introduced in the Application class which require the application to be explicitly booted using the [boot()](https://github.com/laravel/lumen-framework/compare/5.6...5.7#diff-c9248b3167fc44af085b81db2e292837L204) method which is only called from within the [dispatch method](https://github.com/laravel/lumen-framework/compare/5.6...5.7#diff-145186519daafc23b06eb900961a02cfR159) of the RoutesRequests trait.

I would submit a patch for this but I'm not sure on where to call the boot method for a testing scenario. It seems like a good place do to so would be the refreshApplication() method of the TestCase class.

### Steps To Reproduce:
1. Create a new test.
```php
<?php

use Illuminate\Auth\GenericUser;
use Test\TestCase;

class AuthTest extends TestCase
{
    public function setUp()
    {
        parent::setUp();

        $user = new GenericUser([
            'id' => 1
        ]);

        $this->be($user);
    }

    public function testAuth()
    {
         $this->assertTrue(true);
    }
}
```
2. Run `phpunit`

**Comment by georgeboot** [comment]
Did you also encounter #806? I had that issue first, and then after my workaround, this one happened.

**Comment by stefpankov** [comment]
I don't have any tests involving models, I would do a quick one to check but am not currently at my desk, if you could, go to the lumen TestCase class and in the refreshApplication method right under [here](https://github.com/laravel/lumen-framework/blob/5.7/src/Testing/TestCase.php#L56) add a call to `$this->app->boot();` and see if that works for both cases and without your workaround.

**Comment by georgeboot** [comment]
@stefpankov I can confirm that workaround indeed works.

**Comment by JacksonIV** [comment]
I made a pull request to fix this issue.

**Comment by georgeboot** [comment]
Fixed in #809
